### PR TITLE
Remove room

### LINKED ISSUE #37: Remove room
**Is your feature request related to a problem? Please describe.**
At the moment rooms can only be created, accessed and managed. There is no way to remove a room.

**Describe the solution you'd like**
Remove a room if requested and confirmed by owner. If a meeting of the room is still running, force end to prevent bypass of room limit.

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by SamuelWei**
# Fixes #37

## Type (Highlight the corresponding type)
- Bugfix
- **Feature**
- Documentation
- Refactoring (e.g. Style updates, Test implementation, etc.)
- Other (please describe):

## Checklist
- [x] Code updated to current masters head
- [x] Passes CI checks
- [x] Is a part of an issue
- [x] Tests added for the bugfix or newly implemented feature, describe below why if not
- [x] Changelog is updated
- [x] Documentation of code and features exists

## Changes

- Added component to remove a room with confirmation and basic customising 
- Improve behavior of `can `and `cannot `for policy changes

### CODE REVIEW

File: resources/js/views/rooms/View.vue [hunk 1]
```diff
@@ -1,6 +1,16 @@
 <template>
   <div class="container mt-5 mb-5" v-cloak>
     <template v-if="room">
+
+      <!-- Delete button and modal -->
+      <delete-room-component
+        @roomDeleted="$router.push({ name: 'rooms.index' })"
+        v-if="room.isOwner"
```
Comment (dennis95stumm) [inline resources/js/views/rooms/View.vue:8]: This could be solved using policies in the frontend

File: resources/js/lang/de/rooms.js [hunk 2]
```diff
@@ -36,6 +36,14 @@ export default {
       title: 'Zugriff verweigert'
     }
   },
+  modals: {
+    delete: {
+      title: 'Raum löschen',
+      confirm: 'Soll der Raum "{name}" gelöscht werden?',
+      yes: 'Ja',
```
Comment (dennis95stumm) [inline resources/js/lang/de/rooms.js:43]: This translation and the following one should be moved to a central place, to make it dry...

File: tests/Frontend/components/Room/DeleteRoom.spec.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      },
+      attachTo: createContainer()
+    });
+
+    const deleteButton = component.findComponent({ ref: 'deleteButton' });
+    expect(deleteButton.exists()).toBe(true);
+    expect(deleteButton.attributes('disabled')).toBe('disabled');
+  });
+
+  it('successfull delete', function (done) {
```
Comment (dennis95stumm) [inline tests/Frontend/components/Room/DeleteRoom.spec.js:68]: A test for an error response and a test for room where the user is not the owner would be nice!

**Review by dennis95stumm — CHANGES_REQUESTED** [review]

File: resources/js/policies/RoomPolicy.js [hunk 4]
```diff
@@ -10,6 +10,10 @@ export default {
    */
   create (permissionService) {
     return permissionService.currentUser && permissionService.currentUser.permissions.includes('rooms.create');
+  },
+
+  delete (permissionService, model) {
+    return model.isOwner || permissionService.currentUser.permissions.includes('rooms.delete');
```
Comment (dennis95stumm) [inline resources/js/policies/RoomPolicy.js:16]: Shouldn't the permission be created in the seeder?

Comment (SamuelWei) [reply resources/js/policies/RoomPolicy.js:16]: Good point

**Review by dennis95stumm — APPROVED** [review]

**Pull request merged**